### PR TITLE
fix: add link to `nodejs.org/download/release`

### PR DESCRIPTION
**Pull request by aduh95**
## Description

I was looking for that link (which was present on the old website), so I figured I might as well add it back.

## Validation

Who's got time for that, am I right?

### Check List

- [ ] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [ ] I have run `npm run format` to ensure the code follows the style guide.
- [ ] I have run `npm run test` to check if all tests are passing.
- [ ] I have run `npx turbo build` to check if the website builds without errors.
- [ ] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: pages/en/download/prebuilt-binaries/index.mdx [hunk 1]
```diff
@@ -13,12 +13,12 @@ I want the <Release.VersionDropdown /> version of Node.js for <Release.Operating
...
+Read the changelog for <Release.ChangelogLink>this version</Release.ChangelogLink>.
 
-Read the blog post for <Release.BlogPostLink>this version</Release.BlogPostLink>
+Read the blog post for <Release.BlogPostLink>this version</Release.BlogPostLink>.
 
-Learn how to <Release.VerifyingBinariesLink>verify signed SHASUMS</Release.VerifyingBinariesLink>
+Learn how to <Release.VerifyingBinariesLink>verify signed SHASUMS</Release.VerifyingBinariesLink>.
 
-Check out <LinkWithArrow href="https://nodejs.org/download/nightly/">Nightly</LinkWithArrow> prebuilt binaries or <LinkWithArrow href="https://unofficial-builds.nodejs.org/download/">Unofficial Builds</LinkWithArrow> for other platforms
+Check out <LinkWithArrow href="https://nodejs.org/download/nightly/">Nightly</LinkWithArrow> prebuilt binaries, all <LinkWithArrow href="https://nodejs.org/download/release/">Release</LinkWithArrow> prebuilt binaries, or <LinkWithArrow href="https://unofficial-builds.nodejs.org/download/">Unofficial Builds</LinkWithArrow> for other platforms.
```
Comment (ovflowd) [inline pages/en/download/prebuilt-binaries/index.mdx]: All Release? Not All Releases?

Comment (aduh95) [reply pages/en/download/prebuilt-binaries/index.mdx]: I was thinking all (release) binaries

**Review by bmuenzenmeyer — APPROVED** [review]
Reads okay to me

image

**Review by mikeesto — APPROVED** [review]
I think this is OK to merge too unless @ovflowd has a strong objection. Could you please rebase @aduh95 ?

**Review by aymen94 — APPROVED** [review]

**Comment by ovflowd** [comment]
Sorry that it took so long!

**Pull request merged**